flake.nix

```diff
@@ -183,9 +189,26 @@
         };
         checks = {
           inherit aeneas aeneas-tests
-                  aeneas-verify-fstar
-                  aeneas-verify-coq
-                  aeneas-verify-hol4
-                  aeneas-check-tidiness; };
+            aeneas-verify-fstar
+            aeneas-verify-coq
+            aeneas-verify-hol4
+            aeneas-check-tidiness;
+
+          pre-commit-check = pre-commit-hooks.lib.${system}.run {
```

Comment:
I'm not a huge fan of pre-commit hooks because I rebase a lot... I does make sense though, we can try it for a bit

Reply:
Can you add a comment here to explain what this does (this formats the code at commit time, etc.)?